### PR TITLE
enh: configurable Http status code

### PR DESCRIPTION
**Pull request by huskar-t**

### CODE REVIEW

**Review by sangshuduo — APPROVED** [review]
LGTM

File: httperror/errors.go [hunk 1]
```diff
@@ -4,7 +4,7 @@ const (
 	SUCCESS                      = 0x0
 	TSDB_CODE_RPC_AUTH_FAILURE   = 0x0003
 	HTTP_SERVER_OFFLINE          = 0x1100
-	HTTP_UNSUPPORT_URL           = 0x1101
+	HTTP_UNSUPPORTTED_URL        = 0x1101
```
Comment (sunpe) [inline httperror/errors.go:7]: typo unsupported

File: httperror/errors.go [hunk 2]
```diff
@@ -115,10 +117,26 @@ const (
...
+
+const (
+	RPC_NETWORK_UNAVAIL = 0x000B
+)
+
 var ErrorMsgMap = map[int]string{
 	TSDB_CODE_RPC_AUTH_FAILURE:   "Authentication failure",
 	HTTP_SERVER_OFFLINE:          "http server is not onlin",
-	HTTP_UNSUPPORT_URL:           "url is not support",
+	HTTP_UNSUPPORTTED_URL:        "url is not support",
```
Comment (sunpe) [inline httperror/errors.go:139]: supported

File: controller/rest/error_resp.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func BadRequestResponseWithMsg(c *gin.Context, code int, msg string) {
+	c.AbortWithStatusJSON(http.StatusBadRequest, &Message{
+		Code: code,
+		Desc: msg,
+	})
+	web.SetTaosErrorCode(c, code)
+}
+
```
Comment (sunpe) [inline controller/rest/error_resp.go:43]: to reduce duplicate

```
func UnAuthResponse(c *gin.Context, code int) {
    BadResponse(c, http.StatusUnauthorized, code)
}

func BadRequestResponse(c *gin.Context, code int) {
    BadResponse(c, http.StatusBadRequest, code)
}

func BadResponse(c *gin.Context, httpCode int, code int) {
    errStr, exist := httperror.ErrorMsgMap[code]
    if !exist {
        errStr = "unknown error"
    }
    respWithMsg(c, httpCode, code, errStr)
    web.SetTaosErrorCode(c, code)
}

func respWithMsg(c *gin.Context, httpCode int, code int, msg string) {
    c.AbortWithStatusJSON(httpCode, &Message{
        Code: code,
        Desc: msg,
    })
    web.SetTaosErrorCode(c, code)
}
```

File: controller/rest/error_resp.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+func CommonErrorResponse(c *gin.Context, msg string) {
+	httpCode := getErrorHttpStatus(0xffff)
+	c.AbortWithStatusJSON(httpCode, &Message{
+		Code: 0xffff,
+		Desc: msg,
+	})
+	web.SetTaosErrorCode(c, 0xffff)
+}
```
Comment (sunpe) [inline controller/rest/error_resp.go:93]: func TaosErrorResponse(c *gin.Context, code int, msg string) {
    code = code & 0xffff
    httpCode := getErrorHttpStatus(int32(code))
    respWithMsg(c, httpCode, code, msg)
}

func CommonErrorResponse(c *gin.Context, msg string) {
    httpCode := getErrorHttpStatus(0xffff)
    respWithMsg(c, httpCode, code, msg)
}

**Review by sunpe — APPROVED** [review]
LGTM

**Pull request merged**